src/Classic.jl

```diff
@@ -104,48 +52,47 @@ The actor itself is not accessible directly.
 function spawn end
 
 """
-    function become(source::Actor, target)
+    become(source::Actor, target)
 
 Change the behavior of the `source` actor to `target`.
 
 Behavior change will be effective at the next message processing.
+
+????
 Note that when the behavior of `source` is mutable, then it can be
 mutated from `onmessage` directly. This function can change the type
 of the behavior, and it can also replace immutable behaviors.
+????
 """
 function become end
 
 """
-    function onmessage(me::Actor, msg)
-    function onmessage(me::Actor, msg, ::Racing)
+    onmessage(me::Actor, msg)
 
 Handle the incoming message `msg` received by actor `me`.
 
-The second form will be called for `Racing` messages to
-warn the programmer that the message breaks the actor model and accessing it
-without locking may introduce data races.
-
 Note on async and blocking: `@async` is allowed in `onmessage`, but async code should
 not operate directly on the actor state, only through messages. Blocking operations will
 also work inside `onmessage`, but it is up to the implementation to provide any or no
 concurrency of blocked actors, so blocking should generally avoided if possible.
+"""
+function onmessage end
 
-# Examples
-
-```
-mutable struct Counter
-    counter::Int
-end
-
-struct Increment <: MyMessage end
+"""
+    self()::Addr
 
-function Classic.onmessage(me::Actor{Counter}, ::Increment)
-    behavior(me).counter += 1
-end
-```
+Get the address of your actor. To be called in behavior
+functions.
 """
-function onmessage end
+function self end
 
+#
+# Not sure about the following two!
+#
+# since an actor is not accessible from the outside
+# shouldn't those not be internal functions rather than
+# beeing part of the interface?
+# 
 """
     function behavior(actor::Actor)
```

Comment:
The answer to behavior will be part of the answer to your other question I think, stay tuned! addr can be deleted in favor of self.